Added createdByEmail filter in GiftCard (#13132)
saleor/saleor: CHANGELOG.md

### Other changes
- Expand metric units to support more types of products. - #13043 by @FremahA

> - Fix Error Cannot return null for non-nullable field Webhook.name. - #12989 by @Smit-Parmar
> - Added createdByEmail filter in GiftCard and Test case. - #13132 by @Smit-Parmar
> 

Reviewer: @Smit-Parmar I have slightly redacted the changelog entries for consistency. Also we can move both of them to "GraphQL API" section - they're both strictly related to the API.

```suggestion
- Fix error "Cannot return null for non-nullable field Webhook.name" - #12989 by @Smit-Parmar
- Added `GiftCardFilterInput.createdByEmail` filter - #13132 by @Smit-Parmar
```
